binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/common/AbstractKafkaBinderHealthIndicator.java

```diff
@@ -102,6 +110,74 @@ protected Health safelyBuildTopicsHealth() {
 		}
 	}
 
+	private Health buildTopicsHealth() {
+		try {
+			initMetadataConsumer();
```

Comment:
This change also makes it so that the metadata consumer (and the consumer factory that creates the metadata consumer) is only used by the abstract `buildTopicsHealth`. The binder specific methods do not use the consumer at all. The `initMetadataConsumer` could effectively be private. Not super important but it is a possibility now.